packages/opal-server/opal_server/data/api.py

```diff
@@ -34,58 +47,39 @@ def init_data_updates_router(
 
     @router.get(opal_server_config.ALL_DATA_ROUTE)
     async def default_all_data():
-
         """Look for default data file in the repo clone directory and return its contents."""
         try:
             clone_path = opal_server_config.POLICY_REPO_CLONE_PATH
             data_filename = opal_server_config.POLICY_REPO_DEFAULT_DATA_FILENAME
 
             # Look for data.json in the clone directory
             data_file = Path(clone_path) / data_filename
-            
             if data_file.exists():
```

Comment:
this still seems duplicated with lines 63 and on below. and you could see the outcome is that here you have an optimistic return load_json_data, while down below at line 66 you check for if data: so my suggestion was to refactor, one stage for finding the proper path, and second stage, shared code to load, validate and return. Best, Israel

Reply:
alright i will check that out

Reply:
@iwphonedo  how is it now ?